### PR TITLE
CRUD Comments and Unique Name

### PR DESCRIPTION
**Pull request by limcuili**

### CODE REVIEW

**Review by limcuili — COMMENTED** [review]
Reviewed by Farah, Steph, Marek, and Cui Li

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,15 +1,50 @@
...
+    end
+  end
+
+  # PATCH/PUT /posts/1
+  # PATCH/PUT /posts/1.json
+  def update
+    respond_to do |format|
+      if @post.update(post_params)
+        format.html { redirect_to posts_url, notice: 'Post was successfully updated.' }
+        format.json { render :show, status: :ok, location: @post }
```
Comment (Mschmidt19) [inline app/controllers/posts_controller.rb:30]: format.json is a bit confusing / possibly unnecessary, we should clean that up soon

File: app/views/posts/_form.json.jbuilder [hunk 2]
```diff
@@ -0,0 +1,2 @@
+json.extract! post, :id, :message, :created_at, :updated_at
```
Comment (Mschmidt19) [inline app/views/posts/_form.json.jbuilder:1]: more json stuff D:

File: app/views/posts/new.html.erb [hunk 3]
```diff
@@ -1,6 +1,5 @@
-<%= form_for @post do |form| %>
-  <%= form.label :message %>
-  <%= form.text_field :message %>
+<h1>New Post</h1>
 
-  <%= form.submit "Submit" %>
-<% end %>
+<%= render 'form', post: @post %>
```
Comment (Mschmidt19) [inline app/views/posts/new.html.erb:3]: nice use of form partials

File: spec/features/users_can_sign_up_spec.rb [hunk 4]
```diff
@@ -1,13 +1,26 @@
...
-#     visit '/'
-#     click_link 'Signup'
-#     fill_in 'name', with: 'Mr Sassy'
-#     fill_in 'email', with: 'sassy@makers.com'
-#     fill_in 'password', with: '12345'
-#     click_button 'Submit'
-#     expect(page).to have_content('Signed in as Mr Sassy')
-#   end
-# end
+RSpec.feature 'Main page signup', type: :feature do
```
Comment (Mschmidt19) [inline spec/features/users_can_sign_up_spec.rb:3]: Hooray! Tests!

**Review by Mschmidt19 — APPROVED** [review]
We should clean up our formatting so that we all use the same method of redirecting and flash messages. We should also remove some of the json stuff as they aren't necessary (maybe they will be once we implement React?). Looks good though, none of the issues whould affect our code functionality.

**Pull request merged**